DopplerCustomDomain/CustomDomainProvider/CustomDomainProviderService.cs

```diff
@@ -16,20 +16,28 @@ public CustomDomainProviderService(ILogger<CustomDomainProviderService> logger,
             _consulHttpClient = consulHttpClient;
         }
 
-        public async Task CreateCustomDomain(string domain, string service)
+        public async Task CreateCustomDomain(string domain, string service, TraefikRuleTypeEnum ruleType)
```

Comment:
PUT method should be idempotent, so, applying different rules over the same domain should be the same of applying only the last one. 

One simple way to do it is cleaning the rules before applying the new ones. A more complex approach would be to delete the deprecated one and create the missed ones.

Please, take it into account before merging. Let us talk if I am not being able to explain my idea.